server/controllers/prisonerCellHistoryController.ts

```diff
@@ -0,0 +1,168 @@
+      return Object.entries(groupBy(locationsWithAgencyLeaveDate, 'establishmentWithAgencyLeaveDate')).map(
+        // eslint-disable-next-line no-unused-vars
+        ([key, value]) => {
+          // @ts-expect-error ts-migrate(2339) FIXME: Property 'slice' does not exist on type 'unknown'.
+          const fromDateString = formatTimestampToDate(value.slice(-1)[0].assignmentDateTime)
+          const toDateString = formatTimestampToDate(value[0].assignmentEndDateTime) || 'Unknown'
+
+          return {
+            isValidAgency: !!value[0].establishment,
+            name: value[0].establishment,
+            datePeriod: `from ${fromDateString} to ${toDateString}`,
+            cellHistory: value,
+            key,
+          }
+        },
+      )
+    }
+
+    try {
+      // Parse query params for paging, sorting and filtering data
+      const { clientToken } = res.locals
+      const queryParams: PagedListQueryParams = {}
+
+      const { bookingId, firstName, middleNames, lastName } = prisonerData
+      const name = formatName(firstName, middleNames, lastName, { style: NameFormatStyle.firstLast })
+
+      this.prisonApiClient = this.prisonApiClientBuilder(clientToken)
+
+      const page = 0
+      const cells = await this.prisonApiClient.getOffenderCellHistory(bookingId, { page, size: 10000 })
+      const agencyData = await this.prisonApiClient.getAgencyDetails((await cells).content[page].agencyId)
+      const staff = await Promise.all(
+        (await cells).content.map(cell => this.prisonApiClient.getStaffDetails(cell.movementMadeBy)),
+      )
+
+      const cellData = cells.content.map(cell => {
+        const staffDetails = staff.find(user => cell.movementMadeBy === user.username)
+        const agencyName = cell.agencyId
+        const agency = agencyData
+        const agencyDescription = agency ? agency.description : null
+
+        return {
+          establishment: agencyDescription,
+          location: extractLocation(cell.description, agencyName),
+          isTemporaryLocation: isTemporaryLocation(cell.description),
+          movedIn: cell.assignmentDateTime && formatTimestampToDateTime(cell.assignmentDateTime),
+          movedOut: cell.assignmentEndDateTime && formatTimestampToDateTime(cell.assignmentEndDateTime),
+          assignmentDateTime: moment(cell.assignmentDateTime).format('YYYY-MM-DDTHH:mm:ss'),
+          assignmentEndDateTime: cell.assignmentEndDateTime
+            ? moment(cell.assignmentEndDateTime).format('YYYY-MM-DDTHH:mm:ss')
+            : undefined,
+          livingUnitId: cell.livingUnitId,
+          agencyId: agencyName,
+          movedInBy: formatName(staffDetails.firstName, '', staffDetails.lastName),
+        }
+      })
+
+      const cellDataLatestFirst = cellData.sort((left, right) =>
+        sortByDateTime(right.assignmentDateTime, left.assignmentDateTime),
+      )
+      const currentLocation = cellDataLatestFirst.slice(0, 1)[0]
```

Comment:
Don't really need the `slice`, just get first element `cellDataLatestFirst[0]` - it's non-mutating.